### ISSUE TITLE
Modprobe return error. Exec format error

### ISSUE THREAD

**Issue by DoctorShark** [tomaspinho/rtl8821ce#171]
#modprobe 8821ce
modprobe: ERROR: could not insert '8821ce': Exec format error

# dmesg | grep 8821ce
[    7.448179] 8821ce: version magic '5.8.11-200.fc32.x86_64+debug SMP mod_unload ' should be '5.8.11-200.fc32.x86_64 SMP mod_unload '
[  155.088913] 8821ce: version magic '5.8.11-200.fc32.x86_64+debug SMP mod_unload ' should be '5.8.11-200.fc32.x86_64 SMP mod_unload '

kernel version:
5.8.11-200.fc32.x86_64

Secure boot turned off.

Any advices will be appreciated.

**Comment by sergiomb2** [comment]
hum I think this is an dkms  issue.  dkms for some reason create symbol links in weak kernel directory  (/usr/lib/modules/5.6.16-200.fc31.x86_64/weak-updates/ )  if you delete it you may have not "version magic '5.8.11-200.fc32.x"

```locate 8821ce.ko | grep weak
rm /usr/lib/modules/5.6.15-200.fc31.x86_64/weak-updates/8821ce.ko.xz /usr/lib/modules/5.6.16-200.fc31.x86_64/weak-updates/8821ce.ko.xz /usr/lib/modules/5.7.11-100.fc31.x86_64/weak-updates/8821ce.ko.xz /usr/lib/modules/5.8.10-100.fc31.x86_64/weak-updates/8821ce.ko.xz /usr/lib/modules/5.8.11-100.fc31.x86_64/weak-updates/8821ce.ko.xz /usr/lib/modules/5.8.12-100.fc31.x86_64/weak-updates/8821ce.ko.xz ```

**Comment by frankcrawford** [comment]
I've been chasing this for a different module as well, and it isn't directly a dkms issue, but something with how RedHat adds modules.
Also, aside from removing the files you will need to run "depmod -a <kernel>" to clean it up.

**Comment by sergiomb2** [comment]
> I've been chasing this for a different module as well, and it isn't directly a dkms issue, but something with how RedHat adds modules.
> Also, aside from removing the files you will need to run "depmod -a " to clean it up.

yes ,  https://bugzilla.redhat.com/show_bug.cgi?id=1828455 maybe the best is delete or move /usr/sbin/weak-modules script file ... 

anyway 8821ce will be shipped on kernel 4.9 . https://bugzilla.rpmfusion.org/show_bug.cgi?id=5779#c4

**Comment by sergiomb2** [comment]
just emphasize remove  /usr/sbin/weak-modules  is the best solution for now
